service/controller/chart/v1/resource/release/update.go

```diff
@@ -17,6 +17,12 @@ func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange inte
 	if err != nil {
 		return microerror.Mask(err)
 	}
+	if key.IsDeleted(cr) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("release %#q would be deleted, no need to update it", key.ReleaseName(cr)))
+
+		resourcecanceledcontext.SetCanceled(ctx)
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+	}
```

Comment:
So far we did this in `GetCurrentState` and I would like to keep it consistent in order to reduce complexity. Also, we probably want to return in such a case.

Reply:
@xh3b4sd if I put the checking code in `GetCurrentState` would it work in `delete` event? perhaps CR would have `deletionTimestamp` already then operator stop to work on release deletion.

Reply:
Maybe I do not understand the problem. I just thought you want to cancel on deletion. Or is it that you only want to prevent updates on deletion? This should actually not really happen as the informer and controller should take care.

Reply:
> Or is it that you only want to prevent updates on deletion?  

@xh3b4sd  Yes, this is the case that I want to avoid, but actually it happens in app-operator CI tests. https://2140-161353745-gh.circle-artifacts.com/0/logs/kind-control-plane/containers/chart-operator-7c896db6d6-jmx7z_giantswarm_chart-operator-795f8eff635a230cd617fe7cf9980612b4177f0ed05ebcf2ec77d3bd0d295677.log

Please also look into the PR description, its log indicates that update event is handled after deletion.

Reply:
Ok but that is then a reconciliation issue that we need to test and fix in `operatorkit`. Hacking something here will not help us really.

Reply:
@xh3b4sd I created an issue around it, PTAL https://github.com/giantswarm/giantswarm/issues/7432